src/org/javarosa/core/model/FormDef.java

```diff
@@ -1227,10 +1227,11 @@ public void initialize(boolean newInstance, InstanceInitializationFactory factor
             getLocalizer().setToDefault();
         }
 
-        // TODO: Hm, not 100% sure that this is right. Maybe we should be
-        // using a slightly different event for "First Load" which doesn't
-        // get fired again, but always fire this one?
         if (newInstance) {
+            actionController.triggerActionsFromEvent(Action.EVENT_ODK_INSTANCE_FIRST_LOAD, this);
```

Comment:
Are there any performance issues with firing off two events like this?

Reply:
The weight of triggering an event grows proportionally with the number of listeners there are subscribed. I don't think this extra event has an impact on JR level.

I'd imagine that if Collect reacts to both, then yes, but since it's being deprecated, we can make sure that Collect only listens to the new event.

Reply:
The trigger method loops through actions the form defines to be triggered by the event. Presumably a form creator would use one or the other event name in the form definition but not both. If they used both to trigger the same action there'd be some duplication of effort but inexpensive duplication. And that would be very odd form design. So I'd say no.